src/ir_passes/async.ml

```diff
@@ -252,7 +249,11 @@ let transform prog =
                   unitE()); (* suspend *)
                 ("schedule", varP schedule, (* resume later *)
                   (* try await async (); schedule() catch e -> r(e) *)
-                  (selfcallE [] (ic_replyE [] (unitE())) (varE schedule) (projE (varE vkr) 1))) ]
+                 (let v = fresh_var "call" T.unit in
+                  letE v
+                    (selfcallE [] (ic_replyE [] (unitE())) (varE schedule) (projE (varE vkr) 1))
+                    (check_call_perform_status (varE v) (fun e -> ((projE (varE vkr) 1) -*- e)))))
```

Comment:
`((projE (varE vkr) 1) -*- e)` -> `projE (varE vkr) 1 -*- e`

Reply:
I have trouble understanding the line 255.

Reply:
It checks the global flag, returns unit if successfull and otherwise passes the error e to the failure continuation that throws.

Reply:
Understood!